src/IncludedCommands/Persistent/pbans.luau

```diff
@@ -51,86 +50,275 @@ local function GetCachedUsername(UserId: number): string
     return UsernameCache[UserId]
 end
 
+local function PagesToTable(Pages: BanHistoryPages)
+    local Items = {}
+    while true do
+        table.insert(Items, Pages:GetCurrentPage())
+        if Pages.IsFinished then
+            break
+        end
+        Pages:AdvanceToNextPageAsync()
+    end
+    return Items
+end
 
+type BanHistoryPage = {
+    DisplayReason: string,
+    PrivateReason: string,
+    StartTime: string,
+    Duration: number,
+    Ban: boolean,
+    PlaceId: number,
+}
 
 return {
     Keyword = "pbans",
     Category = "PersistentCommands",
-    Description = "Displays a list of all permanent bans.",
+    Description = "Displays the ban history of a set of players by their user id or username (use user if the name is a number).",
+    Arguments = {
+        {
+            Type = "strings",
+            Name = "Names",
+            Description = "Players to view the ban history of.",
+        },
+    },
     ServerLoad = function(Api: Types.NexusAdminApiServer)
         local PersistentBans = require(script.Parent.Parent:WaitForChild("Resources"):WaitForChild("PersistentBans"))
         Api.CommandData.PersistentBans = PersistentBans.GetInstance(Api)
-        
+
+        local PersistentBans = Api.CommandData.PersistentBans
+
         --Create the remote function.
         local GetPersistentBansRemoteFunction = IncludedCommandUtil:CreateRemote("RemoteFunction", "GetPersistentBans") :: RemoteFunction
-        function GetPersistentBansRemoteFunction.OnServerInvoke(Player)
+        function GetPersistentBansRemoteFunction.OnServerInvoke(Player, Names: {number})
+            --Resolve the names.
+            local Ids = {}
+            for _,Name in Names do
+                for _, Id in PersistentBans:ResolveUserIds(Name) do
+                    table.insert(Ids, Id)
+                end
+            end
             if Api.Authorization:IsPlayerAuthorized(Player, Api.Configuration:GetCommandAdminLevel("PersistentCommands", "pbans")) then
                 if Api.CommandData.PersistentBans:WasInitialized() then
-                    return Api.CommandData.PersistentBans.BansDataStore.Data, CompletedUsernameCache
+                    local PersistentBanHistories = {}
+
+                    for _, Id in Ids do
+                        GetCachedUsername(Id)
+                        local Success, Response = pcall(Players.GetBanHistoryAsync, Players, Id)
+                        if Success then
+                            PersistentBanHistories[Id] = PagesToTable(Response)
+                        else
+                            PersistentBanHistories[Id] = {err = Response}
```

Comment:
I would like `err` to be swapped to `Error` or `ErrorMessage` to be consistent with other keys.